File: app/src/main/java/com/nimtego/plectrum/presentation/di/modules/ActivityModule.java
package com.nimtego.plectrum.presentation.di.modules;

import android.content.Context;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {
    Context context;

    public ActivityModule(Context context) {
        this.context = context;
    }

    @Provides
    public Context context() {
        return this.context.getApplicationContext();
    }
}

File: app/src/main/java/com/nimtego/plectrum/presentation/di/modules/ActivityModule.kt
package com.nimtego.plectrum.presentation.di.modules

import dagger.Provides
import android.app.Activity
import android.content.Context
import dagger.Module


@Module
class ActivityModule internal constructor(context: Activity) {

    private val context: Context

    init {
        this.context = context
    }

    @Provides
    fun context(): Context {
        return context
    }
}